lib/executor/variable_looping_vus.go

```diff
@@ -529,21 +529,21 @@ func (vlv VariableLoopingVUs) Run(ctx context.Context, out chan<- stats.SampleCo
 	activeVUs := &sync.WaitGroup{}
 	defer activeVUs.Wait()
 
-	runIteration := getIterationRunner(vlv.executionState, vlv.logger, out)
-	getVU := func() (lib.VU, error) {
-		vu, err := vlv.executionState.GetPlannedVU(vlv.logger, true)
+	runIteration := getIterationRunner(vlv.executionState, vlv.logger)
+	getVU := func() (lib.InitializedVU, error) {
+		initVU, err := vlv.executionState.GetPlannedVU(vlv.logger, true)
 		if err != nil {
 			cancel()
-		} else {
-			activeVUs.Add(1)
-			atomic.AddInt64(activeVUsCount, 1)
+			return nil, err
 		}
-		return vu, err
+		activeVUs.Add(1)
+		atomic.AddInt64(activeVUsCount, 1)
```

Comment:
Incrementing this WG and counter shouldn't be done here, since the VU isn't active at this point, but I wasn't sure where to move it. I tried moving both the increment and decrement to where `vuHandle.start()` and `gracefulStop()` are called, but no luck with the tests.

Reply:
Likely fixed by 22a76523.